### ISSUE TITLE
Feature development: Estimated data on the map 🗺️ 🔨

### ISSUE THREAD

**Issue by madsnedergaard** [electricitymap/electricitymap-contrib#3575]
## Description

tl;dr: **It's finally time to put a color on the delayed zones (including US)!**

Over the years we have gathered a lot of historical data for zones that are currently grey on the map, because the data is too delayed. 

In the last year we've been working hard on the backend to put infrastructure and modelling in place to infer a high-quality real-time value and finally do justice to the hard work that's been done on on verifying and implementing data sources for these zones.

### Current status

First prototype is live here: https://app.electricitymap.org/map?feature=estimations - please check it out and write your thoughts/ideas as comments here :)

---

### An open process

electricityMap is build by you and us together, and we want to take that further 💪 

_Therefore the design and implementation of this feature will be open and transparent for everyone in the community!_

In this issue we will share how we're going to do it, why we're doing it and share progress as well as ask for feedback and testing of early prototypes from you all. We sincerely hope you'll help us by sharing your inputs, ideas and help test the early iterations 💚 

## Thoughts behind this feature

In the next decade, the transition to a low-carbon electricity system will require every organisation and individual to understand and reduce their real electricity footprint. We believe our data and the information from the app is highly relevant to support this transition. 

However, it's not all zones in which we are currently able to provide this information at every hour of the day. For some cases, the data provider has an outage, other data providers may have long delays of several days (or in some cases several months). Then there are those zones in which there simply aren't any hourly data at all.

All of this makes it more difficult, or even impossible, to understand and reduce the real electricity footprint in all parts of the world without further action. 

In response, the electricityMap team has over the last year worked on providing high quality estimations in the API that is able to cover the gaps and delays from data providers, and in some cases, even estimate the carbon intensity for countries with aggregated or consolidated data. 

So far, the results have been really good and we are super excited to finally bring this data to the app.

**Comment by madsnedergaard** [comment]
## The roadmap

#### Version 0.1

**Goal:** Show estimated carbon-intensity for zones that have delayed data

_Features:_

- Delayed countries will be colored on the map!
- Estimated carbon intensity, low-carbon and renewable percentages will be shown
- Production breakdown, imports and exports will be disabled for this first iteration
- The estimation model will be very simple in the first iteration, then switched to more advanced models later

#### Version 0.2

**Goal:** Expand to show all estimated data + better estimation models

_Features:_
- Enabled graphs and production breakdown with estimated data
- Implements feedback for contributors
- Improved estimation models
- Add support for filing holes/gaps in the data with estimated data

#### Version 0.3

**Goal:** Support for completely new zones with aggregated or consolidated data sources

_Features:_
- This means we can get zones like South Korea, Hong Kong and Indonesia filled in!
- Better methodology documentation and potentially support for different estimation models

#### Version 0.4

**Goal:** TBD

### How you can contribute

We will post some initial prototypes and more concrete task/issues, but in the meantime we'd love to hear any thoughts or suggestions you may have!

**Comment by Kongkille** [comment]
I have an initial UI prototype for version 0.1

[Here's the link for the simple interactive prototype.](https://www.figma.com/file/zngVmBt4c1FuGg1IK9cNaI/Estimation-zones?node-id=0%3A1)

The steps in the prototype are as follows

1. Hover over the region with the red outline -> tooltip appears
2. Click the region -> the country panel pops up with estimated data
3. Drag the timeslider, here we see that the country panel goes back to the time where there are measured data.

Alternatively you can also use the arrow keys to skip through the stages. This would resemble our current UI quite a bit. The zones are colored on the map using the inferred data, and then when the user enters the country panel, more information becomes available.

I'm also adding a still-image here:
![image](https://user-images.githubusercontent.com/45588799/143880067-4b020e6e-c0be-478f-bfdf-d91ef4689925.png)


Let me know your thoughts, is this easy to understand, any suggestions or comments?

**Comment by jarek** [comment]
Exciting news!

Do you have any plans for handling regions with only monthly or even yearly data available, but which are overwhelmingly one generation source?

For example, [Manitoba](https://www150.statcan.gc.ca/t1/tbl1/en/tv.action?pid=2510001501&pickMembers%5B0%5D=1.8&pickMembers%5B1%5D=2.2&cubeTimeFrame.startMonth=04&cubeTimeFrame.startYear=2020&cubeTimeFrame.endMonth=11&cubeTimeFrame.endYear=2021&referencePeriods=20200401%2C20211101) has between April 2020 and August 2021 generated between 96% and 98% of its electricity from hydro, 1.6% to 3.6% from wind, and 0.0% to 0.12% from fossil fuels. There's no hourly data, but the fossil numbers are really really low. Or for another example, British Columbia looks similar, but their hydro varies between 90% and 95%, and fossil generation goes up to 6.9% of the total in some months.

Would that fit in your v0.3 goal?

**Comment by Kongkille** [comment]
> Do you have any plans for handling regions with only monthly or even yearly data available, but which are overwhelmingly one generation source?

Yes, exactly. Manitoba sounds like a good candidate for the estimations in v0.3. A nice addition to this is also that as we add more and more grids, we'll also get a more complete import breakdown, which further increases the accuracy of all other grids. In v0.1 we are not considering interconnectors, but it will most likely be part of it in v0.3 (or v0.4) :)

**Comment by madsnedergaard** [comment]
I have created a first draft of the wiki page describing our estimation methodology: https://github.com/electricityMap/electricitymap-contrib/wiki/Estimation-methods

It's not very detailed as this estimation model is mostly going to be used while we get this feature rolled out, but feel free to make changes to the document if it can be improved :)

**Comment by madsnedergaard** [comment]
Sorry for the lack of info here - this should have been released 13 days ago, but unfortunately our current build pipeline is currently being blocked because of some other issues. We're working on fixing it (hopefully done by today or tomorrow), but Christmas and Covid-19 makes planning things harder than usual :)

**Comment by madsnedergaard** [comment]
**It's finally live: https://app.electricitymap.org/map?feature=estimations**
Would love some early feedback and opinions - is it useful and good enough, or how do you feel about it? :)

**Comment by steren** [comment]
My only feedback is that I was expecting the estimated datapoint to appear on the chart, and ideally when hovering it, I expected to see a breakdown by source of the estimation
  
![image](https://user-images.githubusercontent.com/360895/147792817-f5106164-bb01-49e0-b3aa-e66749efe733.png)

**Comment by steren** [comment]
It could also be relevant to call out on country hover and in the panel that the data point is estimated: 
![image](https://user-images.githubusercontent.com/360895/147792866-7733630f-a747-4421-b72b-8c384f7adfbb.png)

This could be achieved by adding a small "(estimated)" under the carbon intensity number

**Comment by madsnedergaard** [comment]
Thanks for the feedback @steren (and @tth4vb + other) on Twitter!
We also discussed the label before releasing, and couldn't figure out if it adds value - so thanks for bringing it up :)
Here's a PR that addresses this: #3664

**Comment by madsnedergaard** [comment]
We're actually kinda done with the initial roadmap here, so I think it's time to close this issue 🥳 
- Actual estimations are live and [documented here](https://github.com/electricityMap/electricitymap-contrib/wiki/Estimation-methods)
- Web app and iOS shows estimations, Android release is in progress

We're going to expand the list of estimated zones over time, so we'd still love your help on gathering information about available historical data for this 💪
